### PR TITLE
feat: [CDS-76926]: create api to get deploy count from start date for ng

### PR DESCRIPTION
**Pull request by rafaelmart1ns**
## Describe your changes
Create an API to return the count of deploy in ng.
This query that is used in this pr was suggested for dbre. The motivation is that very soon Timescale will be removed and the hypertables will be replaced with Postgres partitioned tables: any references to Timescale metadata objects will fail at that point.  This script does not rely on Timescale metadata and works with and without Timescale.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52383)

### CODE REVIEW

**Comment by RajBaviskar** [comment]
trigger uts

**Comment by lasdox** [comment]
How will UI know the previous cont before getting the delta?

**Comment by rafaelmart1ns** [comment]
> How will UI know the previous cont before getting the delta?

Not need to use delta here, I tested on prod as read-only using this request, and its working with timeout.

**Comment by lasdox** [comment]
> > How will UI know the previous cont before getting the delta?
> 
> No need to use delta here, It's using the timescaledb and having the start time to get from date.

Please make sure we are using estimates and not actually counting the entire index tree

File: 125-cd-nextgen/src/main/java/io/harness/cd/TimeScaleDAL.java [hunk 1]
```diff
@@ -468,6 +479,71 @@ public Integer getTotalEnvCount(String accountIdentifier, Table<Record2<String,
...
+                                 .on(field(name("i", "inhparent")).eq(field(name("t2", "oid"))))
+                                 .join(table(name("pg_catalog", "pg_namespace")).as(unquotedName("t3")))
+                                 .on(field(name("t2", "relnamespace")).eq(field(name("t3", "oid"))))
+                                 .join(table(name("pg_catalog", "pg_class")).as(unquotedName("t4")))
+                                 .on(field(name("i", "inhrelid")).eq(field(name("t4", "oid"))))
+                                 .join(table(name("pg_catalog", "pg_namespace")).as(unquotedName("t5")))
+                                 .on(field(name("t4", "relnamespace")).eq(field(name("t5", "oid"))))));
+
+    try {
+      BigDecimal a =
```
Comment (lasdox) [inline 125-cd-nextgen/src/main/java/io/harness/cd/TimeScaleDAL.java:525]: what is "a"? Can we give a more meaningful name if possible please?

Comment (rafaelmart1ns) [reply 125-cd-nextgen/src/main/java/io/harness/cd/TimeScaleDAL.java]: done

**Comment by rafaelmart1ns** [comment]
trigger uts

**Comment by rafaelmart1ns** [comment]
trigger uts

**Review by lasdox — DISMISSED** [review]

**Comment by lasdox** [comment]
By using guava cache, each pod will have a different cache.
Is that acceptable? please confirm

**Review by lasdox — APPROVED** [review]

**Review by Fernando-Dourado — APPROVED** [review]

**Comment by rafaelmart1ns** [comment]
trigger uts

**Comment by rafaelmart1ns** [comment]
trigger messagecheck

**Comment by rafaelmart1ns** [comment]
trigger review

**Pull request merged**